src/acl/Acl.cc

```diff
@@ -269,21 +281,12 @@ ACL::ParseAclLine(ConfigParser &parser, ACL ** head)
         new_acl = 0;
     }
 
-    /*
-     * Here we set AclMatchedName in case we need to use it in a
-     * warning message in aclDomainCompare().
-     */
-    AclMatchedName = A->name;   /* ugly */
-
-    A->parseFlags();
-
-    /*split the function here */
-    A->parse();
-
-    /*
-     * Clear AclMatchedName from our temporary hack
-     */
-    AclMatchedName = nullptr;  /* ugly */
+    CallContextCreator([A] {
+        const auto parseContext = Acl::ParsingContext::Pointer::Make(A->name);
```

Comment:
The ACL context is not created by parseFlags() and does not end with parse(). If possible, please create parseContext earlier, when the ACL name becomes known.

To reduce diff, I suggest trying calling CodeContext::Reset(parseContext) directly, without any CodeContext wrappers. This suggestion relies on the parse_line() change suggested elsewhere. If this works, we may add another wrapper to accommodate parsing code like this one.